src/design-pages/passwordless_gdm.rst

```diff
@@ -0,0 +1,504 @@
+Passwordless-GDM authentication integration
+=======================================================
+
+Related ticket(s):
+------------------
+ * https://github.com/SSSD/sssd/issues/7069
+
+Problem statement
+-----------------
+Passwordless authentication is becoming increasingly popular. SSSD and FreeIPA
+already provide several authentication mechanisms which make use of it:
+`Smart Cards <https://sssd.io/design-pages/smartcards.html>`__,
+`External Identity Providers
+<https://freeipa.readthedocs.io/en/latest/designs/external-idp/external-idp.html>`__
+(EIdP) and `Passkeys <https://sssd.io/design-pages/passkey_kerberos.html>`__.
+Unfortunately, the integration of these mechanisms into the graphical interface
+leaves much to be desired. Some of them may work in a degraded mode, while
+others can’t be used at all.
+
+SSSD and the GUI should be better integrated to make all these authentication
+mechanisms effortless for the user. This would increase the overall security of
+the system, by providing the benefits of these authentication mechanisms, i.e.
+passwordless, MFA, etc.
+
+SSSD and `GDM <https://wiki.gnome.org/Projects/GDM>`__ are working together to
+provide a set of interfaces that can be used to enable these authentication
+mechanisms in Linux’s GUI. While the initial work targets SSSD-GDM integration,
+the objective is that these interfaces can be used by any other desktop
+environment.
+
+Use cases
+---------
+* As a centrally managed user, I want to choose the authentication mechanism
+  to login from the graphical interface so that I can select the one that best
+  suits my needs.
+
+* As a centrally managed user, I want to use an external identity provider
+  (IdP) to login from the graphical interface so that the user interface is
+  easy to use and consistent across all authentication mechanisms in the
+  distribution.
+
+* As a centrally managed user, I want to select the smart card identity to
+  login from the graphical interface so that the authentication is performed
+  with the correct credentials.
+
+* As a centrally managed user, I want to use a passkey to login from the
+  graphical interface so that the user interface is easy to use and consistent
+  across all authentication mechanisms in the distribution.
+
+* As a centrally managed user, I want to get notified when the passkey
```

Comment:
Hi,

I think the same is true for the fallback to local Smartcard authentication. Not sure if this is already handled by the current code. If not this might be something to be added in future.

bye,
Sumit

Reply:
Yes, pam_sss informs about this for passkey and it's shown by GDM after authentication and before moving to the "home" screen. If smartcard does the same, then we are covered, if not we should implement it.

In any case, I've added this to the design.